conda/gateways/logging.py

```diff
@@ -253,6 +253,3 @@ def trace(self, message, *args, **kwargs):
 
 logging.addLevelName(TRACE, "TRACE")
 logging.Logger.trace = trace  # type: ignore[attr-defined]
-
-# suppress DeprecationWarning for warn method
-logging.Logger.warn = logging.Logger.warning  # type: ignore[method-assign]
```

Comment:
This _may_ result in additional deprecation warnings, if any of the conda-related packages use the old name, as suggested in https://github.com/search?q=org:conda+lang:python+.warn(&type=code

I would keep it, but deprecate it with our `deprecated.topic` tool. @kenodegard do you think that's appropriate?

Reply:
You are right that this might lead to more deprecation warnings. A refined search (excluding `warnings.warn`, which remains correct, and the conda and conda-build repos, which are dealt with in this PR and #5364 and #5365) [shows matches in 26 files](https://github.com/search?q=org%3Aconda+lang%3Apython+%22.warn%28%22+NOT+%22warnings.warn%28%22+NOT+repo%3Aconda%2Fconda+NOT+repo%3Aconda%2Fconda-build&type=code).

Perhaps we should address those repos as well?

Reply:
I think so, and please remember that no all dependencies are always installing the newest version of something, e.g. by monkeypatching something like that, the conda maintainers have made it really hard to undo it again, unless the minimal conda versions in the other packages are bumped

Reply:
not sure how much added value there is in us also marking this as deprecated when it’s already marked as deprecated in stdlib

plus marking this as deprecated with our tooling will be unnecessarily complex (our tooling assumes we are deprecating stuff within our own modules, not deprecating monkeypatches from other modules)

+1 on doing a best effort to fine any `logging.warn` in conda and conda-incubator projects and updating those

Reply:
Those in `conda` are fixed (see linked PRs in description), surprisingly, there doesn't seem to be anything in `conda-incubator`, see https://github.com/search?q=org%3Aconda-incubator+lang%3Apython+%22.warn%28%22&type=code.

Reply:
Thanks @kenodegard!